core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -169,16 +181,21 @@ public FileIO io() {
     return io;
   }
 
-  private TableMetadata updateCurrentMetadata(LoadTableResponse response) {
-    // LoadTableResponse is used to deserialize the response, but config is not allowed by the REST
-    // spec so it can be
-    // safely ignored. there is no requirement to update config on refresh or commit.
-    if (current == null
-        || !Objects.equals(current.metadataFileLocation(), response.metadataLocation())) {
-      this.current = response.tableMetadata();
+  private void updateCurrentMetadata(LoadTableResponse response) {
```

Comment:
I don't think this change makes sense. This keeps the original method name, but rather than delegating responsibility to the method, it makes this just a wrapper around a single-line update. Now the method has no purpose, but the same name -- which is misleading. Changes like this can lead to bugs. I recommend not changing the way methods work unless there's a reason to do so.